frontend/src/bundles/common/components/chat-list/chat-list.tsx

```diff
@@ -0,0 +1,102 @@
+import { useCallback, useEffect, useState } from '../../hooks/hooks.js';
+import { Grid } from '../components.js';
+import { ChatListItem, ChatListSearch } from './components.js';
+import styles from './styles.module.scss';
+
+const EMPTY_ARRAY = 0;
+
+type ChatListItemType = {
+    userId: string;
+    username: string;
+    lastMessage?: string;
+    lastMessageDate?: string;
+    avatar?: string;
+    itemSelected?: boolean;
+};
+
+type Properties = {
+    chatItems: ChatListItemType[];
+};
+
+const getItemsWithSelected = (
+    items: ChatListItemType[],
+    id: string,
+): ChatListItemType[] => {
+    return items.map((item) => {
+        if (item.userId !== id) {
+            item.itemSelected = false;
+            return item;
+        }
+        item.itemSelected = true;
+        return item;
+    });
```

Comment:
Avoid such mutations like in this case. Change it to: 
```suggestion
    return items.map((item) => ({
        ...item,
        itemSelected: item.userId === id
    }));
```

Reply:
Move it also to helpers

Reply:
I don't completely understand adding helpers logic, some pathes lead to shared/build which is gitignored. I add helper into directories but can't properly connect it. 
for example `export { getAvatarInitials } from 'shared/build/index.js';` there is no such directory.